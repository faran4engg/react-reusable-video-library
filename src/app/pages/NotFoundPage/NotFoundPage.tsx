import { Helmet } from 'react-helmet-async';

import { Link } from 'react-router-dom';
import { SvgIcon } from 'app/domains/Common/components/svg-icon';

const NotFoundPage = () => (
  <>
    <Helmet>
      <title>Oops ! Error</title>
      <meta name="description" content="not found page" />
    </Helmet>
    <div className="flex flex-col items-center">
      <SvgIcon
        className="mt-8 w-16 h-16 text-theme-color-light dark:text-theme-color-dark"
        icon="Error404Icon"
      />

      <h1 className="text-6xl font-semibold text-gray-700 dark:text-gray-200">
        404
      </h1>
      <p className="my-4 text-gray-700 dark:text-gray-300">
        Page not found. Check the address or
      </p>

      <Link
        className="animate-bounce text-theme-color-light dark:text-theme-color-dark hover:underline"
        to="/"
      >
        Go to Home page
      </Link>
    </div>
  </>
);

export default NotFoundPage;
