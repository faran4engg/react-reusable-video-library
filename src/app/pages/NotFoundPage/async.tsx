import { lazyLoad } from 'utils/loadable';

export const NotFoundPage = lazyLoad(
  () => import('./NotFoundPage'),
  module => module.default,
);
