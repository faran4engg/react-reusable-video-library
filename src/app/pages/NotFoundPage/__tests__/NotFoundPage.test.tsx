import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import NotFoundPage from '..';

const shallowRenderer = createRenderer();

describe('<NotFoundPage />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<NotFoundPage />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
