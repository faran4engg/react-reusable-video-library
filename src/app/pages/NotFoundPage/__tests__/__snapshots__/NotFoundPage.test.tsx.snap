// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render and match the snapshot 1`] = `
<React.Fragment>
  <Helmet>
    <title>
      Oops ! Error
    </title>
    <meta
      content="not found page"
      name="description"
    />
  </Helmet>
  <div
    className="flex flex-col items-center"
  >
    <SvgIcon
      className="mt-8 w-16 h-16 text-theme-color-light dark:text-theme-color-dark"
      icon="Error404Icon"
    />
    <h1
      className="text-6xl font-semibold text-gray-700 dark:text-gray-200"
    >
      404
    </h1>
    <p
      className="my-4 text-gray-700 dark:text-gray-300"
    >
      Page not found. Check the address or
    </p>
    <Link
      className="animate-bounce text-theme-color-light dark:text-theme-color-dark hover:underline"
      to="/"
    >
      Go to Home page
    </Link>
  </div>
</React.Fragment>
`;
